src/spec/dsl.cr

```diff
@@ -110,10 +110,24 @@ module Spec
   end
 
   # :nodoc:
-  def self.matches?(description, file, line, end_line = line)
+  def self.add_tag(tag)
+    set =
+      if tag.starts_with?("~")
```

Comment:
I think this can be replaced with:

```crystal
if antitag = tag.lchop?('~')
  tag = antitag
  @@anti_tags ||= Set(String).new
else
  @@tags ||= Set(String).new
end
```